packages/hydra-indexer/src/substrate/SubstrateService.ts

```diff
@@ -52,6 +48,33 @@ export class SubstrateService implements ISubstrateService {
     })
   }
 
+  private async connect() {
+    this.api = await createApiPromise()
+
+    const subscriptions = this.subscribeToHeads(this.api)
+
+    this.api
+      .once('disconnected', () => {
```

Comment:
I experimented with listening for this event, but it never seems to fire (or it is firing earlier than we are able to capture it). Note even some of the commented code I saw seemed to suggest a similar experience by early developers. https://github.com/Joystream/hydra/blob/0a91b97634af704c42d54dffd7f82f42a8bef2f6/packages/hydra-indexer/src/substrate/SubstrateService.ts#L41 (expecting the API_CONNECTED event to be emitted here https://github.com/Joystream/hydra/blob/0a91b97634af704c42d54dffd7f82f42a8bef2f6/packages/hydra-indexer/src/substrate/index.ts#L63

Reply:
I remember seeing some comments in polkadot/api code that handles connections that suggested why this is the case.. will try to share link to code.

Reply:
> But it is more likely because we are using async/await pattern and awaiting on the api.isReadyError which is capturing the connected event

Correct, I confirmed that, you can listen to that event on provider instance directly, e.g

```ts
 const provider = new WsProvider(conf.WS_PROVIDER_ENDPOINT_URI)
 provider.on('connected', () => {
   console.log('connected')
 })

```